modules/@apostrophecms/schema/index.js

```diff
@@ -964,7 +821,7 @@ module.exports = {
           field.withType = withType;
         }
         if (!field.idsField) {
-          fail('idsField property is missing. Hint: joinByArray takes idsField, NOT idField.');
+          fail('idsField property is missing. Hint: join takes idsField, NOT idField.');
```

Comment:
You can remove this dead code check, since there is logic to automatically set idsField (see above).